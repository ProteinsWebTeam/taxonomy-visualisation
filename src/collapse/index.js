const collapse = node => {
  if (typeof node !== 'object' || Array.isArray(node)) {
    throw new Error('"collapse" function was not passed a valid tree');
  }
  if (node.children) {
    node._children = node.children;
    node.children = null;
    node._children.forEach(collapse);
  }
};

export default collapse;
