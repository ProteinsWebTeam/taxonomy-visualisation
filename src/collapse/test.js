import collapse from '.';

test('Should collapse when passed valid trees', () => {
  const trees = [
    {},
    { children: [{}] },
    { children: [{}, {}] },
    { children: [{ children: [{}, {}] }, { children: [{}] }] },
  ];
  for (const tree of trees) {
    collapse(tree);
    expect(tree).toMatchSnapshot();
  }
});

test('Should fail when passed invalid trees', () => {
  const notTrees = [undefined, null, 1, 'string', []];
  for (const notTree of notTrees) {
    expect(() => collapse(notTree)).toThrow();
  }
});
