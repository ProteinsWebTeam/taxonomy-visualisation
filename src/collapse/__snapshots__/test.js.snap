// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should collapse when passed valid trees 1`] = `Object {}`;

exports[`Should collapse when passed valid trees 2`] = `
Object {
  "_children": Array [
    Object {},
  ],
  "children": null,
}
`;

exports[`Should collapse when passed valid trees 3`] = `
Object {
  "_children": Array [
    Object {},
    Object {},
  ],
  "children": null,
}
`;

exports[`Should collapse when passed valid trees 4`] = `
Object {
  "_children": Array [
    Object {
      "_children": Array [
        Object {},
        Object {},
      ],
      "children": null,
    },
    Object {
      "_children": Array [
        Object {},
      ],
      "children": null,
    },
  ],
  "children": null,
}
`;
